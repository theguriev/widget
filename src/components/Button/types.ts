import { HTMLAttributes, ReactNode } from 'react';

export type ButtonProps = HTMLAttributes<HTMLButtonElement> & {
  variant?: 'circle' | 'list';
  children?: ReactNode;
};
