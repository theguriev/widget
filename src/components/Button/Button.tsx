import React, { FC } from 'react';
import { joinClasses } from '@/utils';
import { ButtonProps } from './types';
import styles from './styles.module.css';

export const Button: FC<ButtonProps> = ({
  variant = 'circle',
  children,
  className,
  ...rest
}) => (
  <button
    type="button"
    className={joinClasses(styles[variant], className || '')}
    {...rest}
  >
    {children}
  </button>
);
