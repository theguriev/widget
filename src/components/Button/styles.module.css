.list,
.circle {
  border: 0;
  background: var(--chicago);
  box-shadow: var(--shadow-button);
  cursor: pointer;
}

.circle {
  border-radius: 100%;
  width: 50px;
  height: 50px;
  padding: 14px;
}

.list {
  border-radius: 24px;
  padding: 14px 24px;
  display: flex;
  align-items: center;
  justify-content: center;
}

.circle svg {
  width: 100%;
  height: 100%;
}

.list svg {
  width: 24px;
  height: 24px;
  margin-right: 8px;
}

.list:hover,
.circle:hover {
  background: var(--cairo);
}
