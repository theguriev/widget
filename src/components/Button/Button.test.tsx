import * as React from 'react';
import { render } from '@testing-library/react';

import { Button } from './Button';

describe('Button', () => {
  it('renders Button', () => {
    const { asFragment } = render(<Button />);
    expect(asFragment()).toMatchSnapshot();
  });
});
