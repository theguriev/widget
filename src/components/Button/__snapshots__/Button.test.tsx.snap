// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders Button 1`] = `
<DocumentFragment>
  <button
    class="circle"
    type="button"
  />
</DocumentFragment>
`;
